### DISCUSSION TITLE
Custom scoring function at search time

### DISCUSSION THREAD

**Discussion by fromTechLead** [meilisearch/product#272] [Feedback & Feature Proposal]
Custom scoring function at search time

ElasticSearch analog: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-script-score

For example:
"script_score": {
        "script": {
          "source": "Math.log(2 + doc['my-int'].value)"
        }
      }

**Comment by gmourier** [comment]
Hi @DenisP01 👋

Thanks for this feedback. Could you tell me more about your need? I understand that it would allow you to have a dynamic score for certain searches precisely. 

Why do you need to customize the scoring of the results so precisely for an end-user? What are the moments where it makes sense in your use case?

I am very curious to know more. 🤓

Thanks for your answer!

**Comment by sandstrom** [comment]
This feature request is similar to another one, on decay functions: https://github.com/meilisearch/product/discussions/205

**Comment by macraig** [comment]
Hello everyone 👋

We just released a 🧪 prototype that allows **boosting results that match a custom filter**, and we'd love your feedback.

### How to use the prototype?
See https://www.notion.so/meilisearch/boostingFilter-API-usage-20aae06bc85e41dba828a90331a69f2c

⚠️ We do NOT recommend using this prototype in production. This is for test purposes only.

Feedback and bug reporting when using this prototype are encouraged! Thanks in advance for your involvement. It means a lot to us ❤️

**Reply by sandstrom** [reply]
Thanks for letting us know, but I'd argue that custom filters is different from custom scoring functions.
